### PR TITLE
Add a TOC to CHANGELOG.md.

### PR DESCRIPTION
**Pull request by david-mcmahon**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,26 @@
-# Release Notes for Kubernetes 1.2.0
+<!--THIS TOC NEEDS TO BE GENERATED EVERY TIME THIS FILE IS UPDATED -->
```
Comment (bgrant0607) [inline CHANGELOG.md]: For an example of how to autogenerate a TOC, see https://raw.githubusercontent.com/kubernetes/kubernetes/master/docs/devel/api_changes.md

Comment (david-mcmahon) [inline CHANGELOG.md]: Oh, I just wrote this quickly to generate it.  The "GENERATED_TOC" comment in the link you sent seems to indicate something does this automagically somehow/somewhere.  I'll take a look.
```
#!/bin/bash
#
# Markdown TOC
#
# Throws out toplevel (#) heading assuming it's the title
firstheading="##"
if [[ "$1" == --include-title ]]; then
  firstheading="#"
  shift
fi
file=$1

echo "Table of Contents"
echo "================="
while read level heading; do
  indent="$(echo $level |sed -e "s,^$firstheading,,g" -e 's,#,  ,g')"
  # fix anchor
  anchor=${heading,,}
  anchor=${anchor// /-}
  anchor=${anchor//[\?\*\,\/()]/}
  echo "$indent* [$heading](#$anchor)"
done < <(egrep "^$firstheading" $file)
```

Comment (bgrant0607) [inline CHANGELOG.md]: Just include the following:
```
<!-- BEGIN MUNGE: GENERATED_TOC -->
<!-- END MUNGE: GENERATED_TOC -->
```
and run hack/update-generated-docs.sh

Comment (bgrant0607) [inline CHANGELOG.md]: Hmm, ok, looks like you need to add munging of this doc:
https://github.com/kubernetes/kubernetes/blob/master/hack/after-build/update-generated-docs.sh#L42

Which would get us analytics etc, as well.

**Comment by david-mcmahon** [comment]
PTAL

**Comment by bgrant0607** [comment]
LGTM, though e2e failed and other tests haven't completed yet

**Pull request merged**